sftp/item.go

```diff
@@ -0,0 +1,80 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/graymeta/stow/local"
+)
+
+type item struct {
+	container *container
+	path      string
+	size      int64
+	modTime   time.Time
+	md        map[string]interface{}
+}
+
+// ID returns a string value that represents the name of a file.
+func (i *item) ID() string {
+	return i.path
+}
+
+// Name returns a string value that represents the name of the file.
+func (i *item) Name() string {
+	return i.path
```

Comment:
hmmm curious if you considered using `filepath.Base` here?

Reply:
We want the item to be the full path minus the container. So if we  have `<container>/<somedir>/item.txt` we want item.Name to be `<somedir>/item.txt` otherwise if you also have `<anotherdir>/item.txt` you can't differentiate between the two.